api/src/main/java/org/geysermc/geyser/api/event/java/ServerCustomPayloadEvent.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2019-2024 GeyserMC. http://geysermc.org
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/Geyser
+ */
+
+package org.geysermc.geyser.api.event.java;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.geysermc.geyser.api.connection.GeyserConnection;
+import org.geysermc.geyser.api.event.connection.ConnectionEvent;
+
+/**
+ * Called when the java server sends a custom payload to geyser.
+ * An extension/addon can use this event to translate this.
+ * This event is not called when a built-in supported custom payload
+ * such as floodgate is received.
+ */
+public class ServerCustomPayloadEvent extends ConnectionEvent {
+
+    private final String channel;
+    private final byte[] data;
+
+    public ServerCustomPayloadEvent(@NonNull GeyserConnection connection, @NonNull String channel, byte[] data) {
+        super(connection);
+        this.channel = channel;
+        this.data = data;
+    }
+
+    /**
+     * The channel that sent this custom payload.
+     * A channel would look something like this:
+     * <code>minecraft:register</code>.
+     * @return the channel that sent this custom payload.
+     */
+    public @NonNull String getChannel() {
```

Comment:
```suggestion
    public @NonNull String channel() {
```
We use fluent accessors in the api, so not get/set prefixes